### PR TITLE
Clang format

### PR DESCRIPTION
**Pull request by crtrott**
Applying clang-format to the repository. Please snap-shot check some files whether this looks good to you or whether there are additional things we want to put into the format file. This is using clang-format 8.0

### CODE REVIEW

**Comment by ndellingwood** [comment]
@crtrott at a glance some of the included header files (at least in the unit tests) were reordered by the tool, wasn't it decided that reordering shouldn't be allowed?

**Comment by crtrott** [comment]
You are right. Also it failed its build ...

**Comment by crtrott** [comment]
Ok I force pushed it now without header reordering. That might fix the build issue before.

**Comment by crtrott** [comment]
Looks like this fixed the previous build issue!

File: algorithms/src/Kokkos_Random.hpp [hunk 1]
```diff
@@ -59,1214 +59,1184 @@
...
+
+    //Default constructor: does not initialize a pool
+    Pool();
+
+    //Initializing constructor: calls init(seed,Device_Specific_Number);
+    Pool(unsigned int seed);
+
+    //Intialize Pool with seed as a starting seed with a pool_size of num_states
+    //The Random_XorShift64 generator is used in serial to initialize all
+  states,
```
Comment (Rombur) [inline algorithms/src/Kokkos_Random.hpp]: This should be on a commented line.  Same for the next few lines

File: .clang-format [hunk 2]
```diff
@@ -0,0 +1,7 @@
+#Official Tool: clang-format version 8.0.0
+BasedOnStyle: google
+SortIncludes: false
+AlignConsecutiveAssignments: true
+AllowShortCaseLabelsOnASingleLine: true
+AllowShortIfStatementsOnASingleLine: true
+#AllowShortLambdasOnASingleLine: true
```
Comment (Rombur) [inline .clang-format]: Why is this commented? If we decide the official tool is clang-format 8.0.0, we can enable this, can't we?

File: algorithms/src/Kokkos_Random.hpp (hunk 1 above)
Comment (crtrott) [reply algorithms/src/Kokkos_Random.hpp]: This is all in a big comment block basically just documentation

File: benchmarks/atomic/main.cpp [hunk 3]
```diff
@@ -1,124 +1,120 @@
...
-    time2 = test_no_atomic<double>(L,N,M,K,R,offsets);
-  if(type==5)
-    time2 = test_no_atomic<Kokkos::complex<double> >(L,N,M,K,R,offsets);
+    double time2 = 1;
+    if (type == 1) time2 = test_no_atomic<int>(L, N, M, K, R, offsets);
+    if (type == 2) time2 = test_no_atomic<long>(L, N, M, K, R, offsets);
+    if (type == 3) time2 = test_no_atomic<float>(L, N, M, K, R, offsets);
+    if (type == 4) time2 = test_no_atomic<double>(L, N, M, K, R, offsets);
+    if (type == 5)
+      time2 = test_no_atomic<Kokkos::complex<double> >(L, N, M, K, R, offsets);
```
Comment (vqd8a) [inline benchmarks/atomic/main.cpp]: @crtrott this should go to the same line of the IF statement, shouldn't it?

Comment (crtrott) [reply benchmarks/atomic/main.cpp]: My guess is this is due to line length issues. At some point you have to get it on a new line. So I think this is fine.

File: .clang-format (hunk 2 above)
Comment (crtrott) [reply .clang-format]: Yeah we can remove that line again I think it is clang-9

Comment (dalg24) [reply .clang-format]: It is an option introduced in Clang 9.  @crtrott please remove the line or add a comment that you will uncomment when we move to Clang-Format version 9.

File: algorithms/src/Kokkos_Random.hpp (hunk 1 above)
Comment (dalg24) [reply algorithms/src/Kokkos_Random.hpp]: I stand with Bruno on this one.  This formatting makes no sense.  Please fix or do `// clang-format off` before the comment block and turn back on after.

File: .clang-format (hunk 2 above)
Comment (crtrott) [reply .clang-format]: I deleted the line

File: algorithms/src/Kokkos_Random.hpp (hunk 1 above)
Comment (crtrott) [reply algorithms/src/Kokkos_Random.hpp]: I turned clang-format off before the comment block and on afterwards.

**Pull request merged**